application/actions/authsocial.go

```diff
@@ -0,0 +1,355 @@
+	}
+}
+
+// Finish the auth callback process for an Orgless user that has no invite associated
+// with this login.
+func socialLoginNonInviteBasedAuthCallback(c buffalo.Context, authEmail, authType, clientID string) error {
+	extras := map[string]interface{}{"authEmail": authEmail, "authType": authType}
+
+	var user models.User
+	if err := user.FindByEmailAndSocialAuthProvider(authEmail, authType); err != nil {
+		return logErrorAndRedirect(c, domain.ErrorGettingSocialAuthUser,
+			fmt.Sprintf("error loading social auth user for '%s' ... %v", authType, err), extras)
+	}
+
+	callbackValues := processSocialAuthCallback(c, authEmail, authType)
+	if callbackValues.errCode != "" {
+		return logErrorAndRedirect(c, callbackValues.errCode, callbackValues.errMsg, extras)
+	}
+
+	authUser, err := getOrglessAuthUser(clientID, user)
+	if err != nil {
+		return err
+	}
+
+	// set person on rollbar session
+	domain.RollbarSetPerson(c, authUser.ID, authUser.Nickname, authUser.Email)
+
+	return c.Redirect(302, getLoginSuccessRedirectURL(authUser, callbackValues.returnTo))
+}
+
+func socialLoginBasedAuthCallback(c buffalo.Context, authEmail, clientID string) error {
+	authType, ok := c.Session().Get(SocialAuthTypeSessionKey).(string)
+	if !ok {
+		return logErrorAndRedirect(c, domain.ErrorMissingSessionSocialAuthType,
+			SocialAuthTypeSessionKey+" session entry is required to complete login")
+	}
+
+	extras := map[string]interface{}{"authEmail": authEmail, "authType": authType}
+
+	// Check for an invite in the Session
+	inviteType, inviteObjectUUID := getInviteInfoFromSession(c)
+
+	// If there is no invite associated with this user, then deal with the User record
+	if inviteObjectUUID == "" {
+		return socialLoginNonInviteBasedAuthCallback(c, authEmail, authType, clientID)
+	}
+
+	// There is an invite associated with this process, so deal ith it
+	callbackValues := processSocialAuthCallback(c, authEmail, authType)
+	if callbackValues.errCode != "" {
+		return logErrorAndRedirect(c, callbackValues.errCode, callbackValues.errMsg, extras)
+	}
+
+	// if we have an authuser, find or create user in local db and finish login
+	var user models.User
+
+	if err := user.FindOrCreateFromOrglessAuthUser(callbackValues.authResp.AuthUser, authType); err != nil {
+		return logErrorAndRedirect(c, domain.ErrorWithAuthUser, err.Error())
+	}
+
+	// If the invite is for a meeting, ensure there is a MeetingParticipant record
+	// that matches the MeetingInvite
+	if inviteType != "" {
+		dealWithInviteFromCallback(c, inviteType, inviteObjectUUID, user)
+	}
+
+	authUser, err := getOrglessAuthUser(clientID, user)
+	if err != nil {
+		return err
+	}
+
+	// set person on rollbar session
+	domain.RollbarSetPerson(c, authUser.ID, authUser.Nickname, authUser.Email)
+
+	return c.Redirect(302, getLoginSuccessRedirectURL(authUser, callbackValues.returnTo))
+}
+
+// Hydrates an AuthUser struct
+func getOrglessAuthUser(clientID string, user models.User) (AuthUser, error) {
```

Comment:
Looks identical to `getAuthUser` except for the initial call to get the access token. Can these be combined into one function?

Reply:
:+1: